### DISCUSSION TITLE
usage of network buffers in USB audio

### DISCUSSION THREAD

**Discussion by fkaab** [zephyrproject-rtos/zephyr#39296] [Q&A]
I'm fairly new to all of this so please forgive me, if some of this should be obvious.

Q: Why does the USB Audio device class driver (under zephyr/subsys/usb/class/audio) use network buffers instead of kernel service buffers?

I looked into some of the code, and none of the Network Buffer API are used really anywhere, except to allocate and unreference the buffers. When underlying functions are called, like usb_device.c/usb_write for example, only some pointers to the buffer data itself and related integers get passed on, instead of the buffer object. Do these buffers have an advantage, or does using them simply not make a difference?

**Comment by koffes** [comment]
I am interested in this as well.
